core/src/test/java/it/pagopa/selfcare/mscore/core/DelegationServiceImplTest.java

```diff
@@ -45,47 +45,70 @@ class DelegationServiceImplTest {
+        dummyDelegationProdPagopa.setTo("taxCodeTo");
         dummyDelegationProdPagopa.setFrom("from");
+
+        dummyDelegationTaxCode = new Delegation();
+        dummyDelegationTaxCode.setToTaxCode("taxCodeTo");
+        dummyDelegationTaxCode.setFromTaxCode("taxCodeFrom");
+        dummyDelegationTaxCode.setProductId("prod-pagopa");
+        dummyDelegationTaxCode.setId("id");
     }
 
     /**
      * Method under test: {@link DelegationServiceImpl#createDelegation(Delegation)}
      */
     @Test
     void testCreateDelegationWithProductProdIo() {
-        Institution institution = new Institution();
-        institution.setId("id");
+        Institution institutionTo = new Institution();
+        institutionTo.setId("idTo");
+        institutionTo.setTaxCode("taxCodeTo");
+        Institution institutionFrom = new Institution();
+        institutionTo.setId("idFrom");
+        institutionTo.setTaxCode("taxCodeFrom");
         when(delegationConnector.save(any())).thenReturn(dummyDelegationProdIo);
+        when(institutionService.retrieveInstitutionById(dummyDelegationProdIo.getFrom())).thenReturn(institutionFrom);
+        when(institutionService.retrieveInstitutionById(dummyDelegationProdIo.getTo())).thenReturn(institutionTo);
         doNothing().when(mailNotificationService).sendMailForDelegation(any(), any(), any());
         doNothing().when(institutionService).updateInstitutionDelegation(any(),anyBoolean());
         Delegation response = delegationServiceImpl.createDelegation(dummyDelegationProdIo);
         verify(delegationConnector).save(any());
         assertNotNull(response);
         assertNotNull(response.getId());
         assertEquals(dummyDelegationProdIo.getId(), response.getId());
+        assertEquals(institutionTo.getTaxCode(), response.getToTaxCode());
+        assertEquals(institutionFrom.getTaxCode(), response.getFromTaxCode());
     }
 
     /**
      * Method under test: {@link DelegationServiceImpl#createDelegation(Delegation)}
      */
     @Test
     void testCreateDelegationForPTWithProductPagopa() {
-        Institution institution = new Institution();
-        institution.setId("id");
-        institution.setInstitutionType(InstitutionType.PT);
+        Institution institutionTo = new Institution();
+        institutionTo.setId("id");
+        institutionTo.setInstitutionType(InstitutionType.PT);
+        institutionTo.setTaxCode("taxCodeTo");
+        Institution institutionFrom = new Institution();
+        institutionFrom.setId("id");
+        institutionFrom.setTaxCode("taxCodeFrom");
         when(delegationConnector.save(any())).thenReturn(dummyDelegationProdPagopa);
         doNothing().when(mailNotificationService).sendMailForDelegation(any(), any(), any());
-        when(institutionService.getInstitutions(any(), any())).thenReturn(List.of(institution));
+        when(institutionService.getInstitutions(any(), any())).thenReturn(List.of(institutionTo));
```

Comment:
you could use eq(..) instead of any()....for ex.  `when(institutionService.getInstitutions(eq(dummyDelegationProdPagopa.getTo()), eq(null)))....`